pkg/plugins/platform/types.go

```diff
@@ -46,7 +49,7 @@ var PluginMap = map[string]App{
 }
 
 // GetPluginApp will return an application of the type specified by pluginType
-func GetPluginApp(pluginType string) (App, error) {
+func GetPluginApp(pluginType string) (*LocalApp, error) {
```

Comment:
Please consider this piece rather heavily. This works best **now** because it is the only way I could add the `InitFromMissingDirectory(string, string)` method on type `LocalApp`, without defining it as a required part of the `App` interface.

In the future, if we deal with other `App` types that are not `LocalApp` we would need to revert to returning the `App` generic. Right now however, the only possible return value is `LocalApp`. I'm not sure what our long term plans are or if there are situations where this may not be the case. Can definitely use feedback here...

Reply:
I'd say don't do it and don't bother with adding the InitFromMissingDirectory(), as elsewhere.